src/VisualStudio/CSharp/Impl/SemanticSearch/SemanticSearchToolWindowImpl.cs

```diff
@@ -0,0 +1,470 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Controls.Primitives;
+using System.Windows.Data;
+using System.Windows.Markup;
+using System.Windows.Media;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Host;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Navigation;
+using Microsoft.CodeAnalysis.Notification;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.SemanticSearch;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Microsoft.VisualStudio.Threading;
+using Microsoft.VisualStudio.Utilities;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.CSharp;
+
+using TextSpan = Microsoft.CodeAnalysis.Text.TextSpan;
+
+[Shared]
+[Export(typeof(ISemanticSearchWorkspaceHost))]
+[Export(typeof(SemanticSearchToolWindowImpl))]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class SemanticSearchToolWindowImpl(
+    IHostWorkspaceProvider hostWorkspaceProvider,
+    IThreadingContext threadingContext,
+    ITextEditorFactoryService textEditorFactory,
+    ITextDocumentFactoryService textDocumentFactory,
+    IContentTypeRegistryService contentTypeRegistry,
+    IVsEditorAdaptersFactoryService vsEditorAdaptersFactoryService,
+    IAsynchronousOperationListenerProvider listenerProvider,
+    IGlobalOptionService globalOptions,
+    VisualStudioWorkspace workspace,
+    IStreamingFindUsagesPresenter resultsPresenter,
+    SVsServiceProvider serviceProvider) : ISemanticSearchWorkspaceHost, OptionsProvider<ClassificationOptions>
```

Comment:
💡 Can remove the `serviceProvider` import since `IVsService<,>` can be used instead